spinalcordtoolbox/scripts/sct_create_mask.py

```diff
@@ -242,23 +242,14 @@ def create_mask(param):
     for iz in range(0, nz, 1):
         if iz in z_centerline_not_null:
             cx[iz], cy[iz] = ndimage.measurements.center_of_mass(np.array(data_centerline[:, :, iz]))
-    # create 2d masks
-    file_mask = 'data_mask'
+
+    # create 2d masks at each axial slice
+    im_out = zeros_like(im_data, dtype=np.float64)
```

Comment:
We use this zero image as a sort of "container" for a concatenation of 2D mask images.

If `im_data` is 4D, however, the shape of this image will wrong for the assignment operation that comes later.

So, I think it would be better to stick to the old method (creating several images then concatenating). But, just don't write images to files.

Reply:
Fixed in cc20933.